### PR TITLE
PHP 8.1 | wpdb::_real_escape(): fix "passing null to non-nullable" deprecation notices (Trac 53635)

### PR DESCRIPTION
**Pull request by jrfnl**
### PHP 8.1 | wpdb::_real_escape(): fix "passing null to non-nullable" deprecation notices

The PHP native `mysqli_real_escape_string()` function expects to be passed a string as the second parameter and this is not a nullable parameter.
Passing `null` to it will result in a `mysqli_real_escape_string(): Passing null to parameter #2 ($string) of type string is deprecated` notice on PHP 8.1.

Previously, an input type check was put in place to prevent fatal errors on PHP 8.0 when an array, object or resource was passed. Changeset [48980].

A `null` value was explicitly excluded from that check, even though a `null` value being passed would only ever result in an empty string anyway.

This commit changes the previous input type check to also bow out early for `null` values and to automatically return an empty string for those.

Refs:
* https://www.php.net/manual/en/mysqli.real-escape-string.php
* https://wiki.php.net/rfc/deprecate_null_to_scalar_internal_arg

Trac ticket: https://core.trac.wordpress.org/ticket/53635

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/db/realEscape.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * @group  wpdb
+ * @covers wpdb::_real_escape
+ */
+class Tests_DB_RealEscape extends WP_UnitTestCase {
+
+	/**
+	 * Test that various types of input passed to `wpdb::_real_escape()` are handled correctly.
+	 *
+	 * Note: this test does not test the actual escaping or other logic in the function.
+	 * It just and only tests and documents how the function handles verious input types.
```
Comment (jrfnl) [inline tests/phpunit/tests/db/realEscape.php:15]:
```suggestion
	 * It just and only tests and documents how the function handles various input types.
```

Comment (jrfnl) [reply tests/phpunit/tests/db/realEscape.php]: Fixed. This PR is now ready for commit.

**Review by hellofromtonya — APPROVED** [review]
[Thoroughly discussed and reviewed this PR during a live working session](https://www.youtube.com/watch?v=BQbaDBDCOIA&t=1858s). This PR is ready for `commit` 👍

**Comment by hellofromtonya** [comment]
Committed in changesets https://core.trac.wordpress.org/changeset/51798 and https://core.trac.wordpress.org/changeset/51799.

**Pull request closed without merging**